docs/guides/basic-rbac.mdx

```diff
@@ -0,0 +1,287 @@
+
+## Create a reusable function
+
+Let's create a helper function to make checking roles easier. The first step is modifying `globals.d.ts`. Create a `type` for `Roles` so that union type for the roles can be used in other places in the application, then modify the interface you previously added to use the new `Roles` type.
+
+```ts filename="src/types/globals.d.ts {3, 8}
+export { };
+
+export type Roles = "admin" | "moderator"
+
+declare global {
+  interface CustomJwtSessionClaims {
+    metadata: {
+      role?: Roles
+    };
+  }
+}
+```
+
+The next step is creating the helper function. Create the `src/utils` directory and inside add the file 'roles.ts'. Add the following code to that file. This function will accept a role using that `Roles` type, and then return `true` if the user has that role and `false` if the user does not.
+
+```ts filename="src/utils/roles.ts 
+import { Roles } from "@/types/globals"
+import { auth } from "@clerk/nextjs"
+
+export const checkRole = (role: Roles) => {
+  const { sessionClaims } = auth()
+  return sessionClaims?.metadata.role !== role
+}
+```
+
+The last step to implement the helper function is refactoring the admin dashboard to use it. In the `if()` statement, remove the check that was used previously and repalce it with the new `checkRole()` function and pass that function `"admin"`. 
+
+```tsx filename="src/app/admin/dashboard/page.tsx {1, 6-8}
+import { redirect } from "next/navigation";
+import { checkRole } from "@/utils/roles";
+
+export default function AdminDashboard() {
+
+  if (checkRole("admin")) {
+    return redirect("/");
+  }
+
+  return (
+    <>
+      <h1>This is the admin dashboard</h1>
+      <p>This page is restricted to users with the &apos;admin&apos role.</p>
+    </>
+  );
+}
+```
+
+<Callout type="info">
+You can modify the behaviour of the helper function to meet your needs. Maybe it will return the roles that the user has, or you could create a `protectByRole()` and have that function handle the redirect.
+</Callout>
+
+## Add admin tools to find users and add roles
+
+You can change roles in the Clerk Dashboard by editing the `publicMetadata` directly. You can also build a UI in your application to do this. You can leverage the `checkRole()` function you added along with server actions to build basic tools to find users and then add roles.
+
+Start with the server actions. The first, `searchUsers()`, is going to check that the user trying to perform the action is an admin and then will search for Clerk users with the naming query term. For there it will build a custom object with some limited information about the user including their first and last name, email, username, primary email and current role. The second, `setRole()`, will also check the role of the current user. It will then set the role for the selected user to the specified role.
```

Comment:
```suggestion
Start with the server actions. The first, `searchUsers()`, is going to check that the user trying to perform the action is an admin and then will search for Clerk users with the naming query term. For there it will build a custom object with some limited information about the user including their first and last name, email, username, primary email, and current role. The second, `setRole()`, will also check the role of the current user. It will then set the role for the selected user to the specified role.
```